### PR TITLE
docker auto-update

### PR DESCRIPTION
**Pull request by casperdcl**
- optional watchtower auto-update
  + closes https://github.com/premAI-io/prem-daemon/pull/121
- fix gateway service naming

### CODE REVIEW

File: docker-compose.gateway.yml [hunk 1]
```diff
@@ -1,18 +1,18 @@
 version: '3.7'
 services:
   
-  premapp:
-    container_name: premapp
+  prem_app:
```
Comment (tiero) [inline docker-compose.gateway.yml:4]: Can you elaborate on why this is necessary?

Comment (casperdcl) [reply docker-compose.gateway.yml:4]: just using the same convention as the base `docker-compose.yml` file :)

Comment (tiero) [reply docker-compose.gateway.yml:4]: oh ok, I think this docker-compose may be outdated and not really used for dev or prod either AFAIK

**Comment by tiero** [comment]
Thank you! this seems to be a better solution than a hand-made solution. 
I think this PR should be opened in `prem-box` as this is more geared toward a production scenario, rather development one.

File: docker-compose.gateway.yml (hunk 1 above)
Comment (casperdcl) [reply docker-compose.gateway.yml:4]: well I use it for dev. @filopedraz shouldn't we use it for `demo.prem.ninja` too?

Comment (tiero) [reply docker-compose.gateway.yml:4]: I think [this service belongs to the actual server installation](https://github.com/premAI-io/prem-app/pull/458#issuecomment-1789076004) for production, you premapp is the one used in the compose

**Comment by casperdcl** [comment]
ah I don't quite understand the use-case of `prem-box`, i.e. why not merge all the confign from `prem-box` into this repo (`prem-app`)?

**Comment by tiero** [comment]
> ah I don't quite understand the use-case of `prem-box`, 

It's the repository dedicated to centralizing installer script, the latest stable versions, and docker-compose files used for the server installation 👉 https://dev.premai.io/docs/prem-app/installation/install-linux-production


> i.e. why not merge all the confign from `prem-box` into this repo (`prem-app`)?

We are going to change the project structure very soon, for now, let's stick with what we have already in place

File: docker-compose.gateway.yml (hunk 1 above)
Comment (filopedraz) [reply docker-compose.gateway.yml:4]: @casperdcl I am using prem-box installer script for demo.prem.ninja :)

**Comment by casperdcl** [comment]
moved to https://github.com/premAI-io/prem-box/pull/33

**Pull request closed without merging**